src/CoreBundle/Bridge/Chameleon/Module/Sidebar/SidebarBackendModule.php

```diff
@@ -0,0 +1,257 @@
+        while ($tableObject = $tableList->Next()) {
+            if (false === $this->isTableAccessAllowed($activeUser, $tableObject)) {
+                continue;
+            }
+            if (false === \array_key_exists($tableObject->fieldCmsContentBoxId, $aMenuItemsTemp)) {
+                $aMenuItemsTemp[$tableObject->fieldCmsContentBoxId] = [];
+            }
+            $aMenuItemsTemp[$tableObject->fieldCmsContentBoxId][] = new MenuItem(
+                $tableObject->fieldTranslation,
+                $tableObject->fieldIconFontCssClass,
+                $this->getTableTargetUrl($tableObject->id)
+            );
+        }
+    }
+
+    private function isTableAccessAllowed(\TCMSUser $activeUser, \TdbCmsTblConf $tableObject): bool
+    {
+        $tableInUserGroup = $activeUser->oAccessManager->user->IsInGroups($tableObject->fieldCmsUsergroupId);
+        $isEditAllowed = $activeUser->oAccessManager->HasEditPermission($tableObject->fieldName);
+        $isShowAllReadonlyAllowed = $activeUser->oAccessManager->HasShowAllReadOnlyPermission($tableObject->fieldName);
+
+        return (true === $tableInUserGroup && (true === $isEditAllowed || true === $isShowAllReadonlyAllowed));
+    }
+
+    private function getTableTargetUrl(string $tableId): string
+    {
+        return PATH_CMS_CONTROLLER."?pagedef=tablemanager&id=$tableId";
+    }
+
+    private function addModuleMenuItems(array &$menuItemsRaw, \TCMSUser $activeUser, string $activeLanguageId): void
+    {
+        $query = "SELECT * FROM `cms_module` WHERE `active` = '1'";
+        $cmsModuleList = \TdbCmsModuleList::GetList($query, $activeLanguageId);
+        while ($cmsModule = $cmsModuleList->Next()) {
+            if (false === $this->isModuleAccessAllowed($activeUser, $cmsModule)) {
+                continue;
+            }
+            if (false === \array_key_exists($cmsModule->fieldCmsContentBoxId, $menuItemsRaw)) {
+                $menuItemsRaw[$cmsModule->fieldCmsContentBoxId] = [];
+            }
+            $menuItemsRaw[$cmsModule->fieldCmsContentBoxId][] = new MenuItem(
+                $cmsModule->fieldName,
+                $cmsModule->fieldIconFontCssClass,
+                $this->getModuleTargetUrl($cmsModule)
+            );
+        }
+    }
+
+    private function isModuleAccessAllowed(\TCMSUser $activeUser, \TdbCmsModule $cmsModule): bool
+    {
+        return true === $activeUser->oAccessManager->user->IsInGroups($cmsModule->fieldCmsUsergroupId);
+    }
+
+    private function getModuleTargetUrl(\TdbCmsModule $cmsModule): string
+    {
+        $url = PATH_CMS_CONTROLLER.'?pagedef='.$cmsModule->fieldModule;
+        if ('' !== $cmsModule->fieldParameter) {
+            $url .= '&'.$cmsModule->fieldParameter;
+        }
+        if ('' !== $cmsModule->fieldModuleLocation) {
+            $url .= '&_pagedefType='.$cmsModule->fieldModuleLocation;
+        }
+
+        return $url;
+    }
+
+    private function mergeMenuItems(array $menuItemsRaw): array
```

Comment:
merge why?

Reply:
The name survived some iterations - changed to `createMenuCategories()`.